latch_cli/services/execute.py

```diff
@@ -1,260 +1,74 @@
-    stdout_channel = kubernetes.stream.ws_client.STDOUT_CHANNEL
-    stderr_channel = kubernetes.stream.ws_client.STDERR_CHANNEL
-
-    class WSStream:
-        def __init__(self):
-            self._wssock = stream(
-                core_v1.connect_get_namespaced_pod_exec,
-                pod_name,
-                namespace,
-                command=["/bin/sh"],
-                stderr=True,
-                stdin=True,
-                stdout=True,
-                tty=True,
-                _preload_content=False,
-            ).sock
-
-        def send(self, chunk: bytes):
-            self._wssock.send(stdin_channel + chunk, websocket.ABNF.OPCODE_BINARY)
-
-        def get_frame(
-            self,
-        ) -> Tuple[int, websocket.ABNF]:
-            return self._wssock.recv_data_frame(True)
-
-        @property
-        def socket(self):
-            return self._wssock.sock
-
-        def close(self):
-            self._wssock.close()
-
-    class TTY:
-        def __init__(
-            self,
-            in_stream: int,
-            out_stream: int,
-            err_stream: int,
-            raw: bool = True,
-        ):
-            if raw:
-                setraw(sys.stdin.fileno())
-
-            self._stdin = in_stream
-            self._stdout = out_stream
-            self._stderr = err_stream
-
-        def flush(self) -> bytes:
-            return os.read(self._stdin, 32 * 1024)
-
-        def write_out(self, chunk: bytes):
-            os.write(self._stdout, chunk)
-
-        def write_err(self, chunk: bytes):
-            os.write(self._stderr, chunk)
-
-        @property
-        def in_stream(self):
-            return self._stdin
-
-    try:
-        old_settings = termios.tcgetattr(sys.stdin.fileno())
-
-        tty_ = TTY(
-            sys.stdin.fileno(),
-            sys.stdout.fileno(),
-            sys.stderr.fileno(),
-        )
-        try:
-            wsstream = WSStream()
-        except kubernetes.client.rest.ApiException:
-            raise ValueError(
-                "Unable to find requested task name - make sure that you are in the correct workspace."
+from pathlib import Path
+from typing import List
+
+from flytekit.core.base_task import PythonTask
+from flytekit.core.context_manager import FlyteEntities
+from scp import SCPClient
+
+from latch_cli.centromere.ctx import CentromereCtx
+from latch_cli.centromere.utils import TmpDir
+from latch_cli.services.utils import import_flyte_objects
+
+
+def execute(local_script: Path):
+    """Executes tasks and workflows on remote servers in their containers."""
+
+    wd = Path().absolute()
+    with CentromereCtx(
+        wd,
+        disable_auto_version=False,
+        remote=True,
+    ) as ctx:
+
+        import_flyte_objects(wd)
+        for entity in FlyteEntities.entities:
+            if isinstance(entity, PythonTask):
+                print(entity.name)
+
+        # nucleus: get latest version associated with task name and later allow
```

Comment:
just ignore this for now and assume i'll get the correct image name, will add sometime today